src/Parallel/DistributedObject.hpp

```diff
@@ -1119,10 +1119,26 @@ template <typename ParallelComponent, typename... PhaseDepActionListsPack>
 void DistributedObject<ParallelComponent,
                        tmpl::list<PhaseDepActionListsPack...>>::
     initiate_shutdown(const std::exception& exception) {
-  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
-  CkError("\nException caught in Algorithm:\n%s\n\nShutting down...\n",
-          exception.what());
-  sys::abort("");
+  // Send message to `Main` that we received an exception and set termination.
+  auto* global_cache = Parallel::local_branch(global_cache_proxy_);
```

Comment:
Add this at the beginning of the function so that we stop performing the current action.
```cpp
performing_action_ = false;
  if constexpr (std::is_same_v<Parallel::NodeLock, decltype(node_lock_)>) {
    node_lock_.unlock();
  }
```